implants/imix/imix_config.yaml.example

```diff
@@ -0,0 +1,182 @@
+# 1. Individual environment variables (e.g., IMIX_CALLBACK_URI)
+# 2. Default values
+#
+# Usage (container-friendly pattern):
+#   Set IMIX_BUILD_CONFIG environment variable with the YAML content:
+#
+#   export IMIX_BUILD_CONFIG="$(cat imix_config.yaml.example)"
+#   cargo build --release
+#
+# Or inline:
+#   export IMIX_BUILD_CONFIG='
+#   callback_uri: "http://tavern.example.com:8000"
+#   callback_interval: 10
+#   '
+#   cargo build --release
+#
+# In Dockerfile:
+#   ENV IMIX_BUILD_CONFIG="callback_uri: http://tavern.example.com:8000\ncallback_interval: 10"
+#
+# In docker-compose.yml:
+#   environment:
+#     IMIX_BUILD_CONFIG: |
+#       callback_uri: "http://tavern.example.com:8000"
+#       callback_interval: 10
+
+# ============================================================================
+# CALLBACKS CONFIGURATION
+# ============================================================================
+# You can specify callbacks in two formats:
+#
+# 1. NEW FORMAT (recommended): List of callbacks with individual settings
+# 2. LEGACY FORMAT: Single callback with global interval/retry settings
+#
+# The new format takes precedence if both are specified.
+
+# --- NEW FORMAT: Multiple Callbacks ---
+# Each callback can have its own URI, interval, and retry_interval
+callbacks:
+  # Primary callback - DNS transport with custom intervals
+  - uri: "dns://8.8.8.8:53?domain=c2.example.com&type=txt"
+    interval: 10          # Check every 10 seconds
+    retry_interval: 5     # Retry after 5 seconds on error
+
+  # Secondary callback - HTTP fallback with different timing
+  - uri: "http://tavern.example.com:8000"
+    interval: 30          # Check every 30 seconds
+    retry_interval: 15    # Retry after 15 seconds on error
+
+  # Tertiary callback - HTTPS with longer intervals
+  - uri: "https://backup.example.com:443"
+    interval: 60          # Check every 60 seconds
+    retry_interval: 30    # Retry after 30 seconds on error
+
+# --- LEGACY FORMAT: Single Callback (for backward compatibility) ---
+# Uncomment these and comment out the 'callbacks' section above to use legacy format
+#
+# callback_uri: "http://tavern.example.com:8000"
+# callback_interval: 10
+# retry_interval: 10
+
+# ============================================================================
+# SERVER CONFIGURATION
+# ============================================================================
+
+# The public key for the tavern server
+# Obtain from server using: curl $IMIX_CALLBACK_URI/status | jq -r '.Pubkey'
+# If not specified, the build will attempt to fetch it automatically
+# Required when using redirectors
+server_pubkey: "your-base64-encoded-server-public-key-here"
+
+# ============================================================================
+# NETWORK CONFIGURATION
+# ============================================================================
+
+# Override system settings for proxy URI over HTTP(S)
```

Comment:
This should exist as an option on http1 and grpc type callbacks and error if set for unsupported callback type.